shared/src/test/diff/qq/NuSyntax.mls

```diff
@@ -27,9 +27,46 @@ run(x `=> x)(42)
 `1 `<= `2
 //│ Code[Bool, nothing]
 
+:p
 `1 `* `2 `+ `3
+//│ |`|1| |`|*| |`|2| |`|+| |`|3|
+//│ AST: TypingUnit(Quoted(App(Var(+), Tup(_: Unquoted(Quoted(App(Var(*), Tup(_: Unquoted(Quoted(IntLit(1))), _: Unquoted(Quoted(IntLit(2))))))), _: Unquoted(Quoted(IntLit(3)))))))
+//│ Parsed: code"+(${code"*(${code"1"}, ${code"2"},)"}, ${code"3"},)";
 //│ Code[Int, nothing]
 
+:p
+`1 `* `2 `+ `3 `- `4
+//│ |`|1| |`|*| |`|2| |`|+| |`|3| |`|-| |`|4|
+//│ AST: TypingUnit(Quoted(App(Var(-), Tup(_: Unquoted(Quoted(App(Var(+), Tup(_: Unquoted(Quoted(App(Var(*), Tup(_: Unquoted(Quoted(IntLit(1))), _: Unquoted(Quoted(IntLit(2))))))), _: Unquoted(Quoted(IntLit(3))))))), _: Unquoted(Quoted(IntLit(4)))))))
+//│ Parsed: code"-(${code"+(${code"*(${code"1"}, ${code"2"},)"}, ${code"3"},)"}, ${code"4"},)";
+//│ Code[Int, nothing]
+
+:p
+`1 `+ `2 `* `3
+//│ |`|1| |`|+| |`|2| |`|*| |`|3|
+//│ AST: TypingUnit(Quoted(App(Var(+), Tup(_: Unquoted(Quoted(IntLit(1))), _: Unquoted(Quoted(App(Var(*), Tup(_: Unquoted(Quoted(IntLit(2))), _: Unquoted(Quoted(IntLit(3)))))))))))
+//│ Parsed: code"+(${code"1"}, ${code"*(${code"2"}, ${code"3"},)"},)";
+//│ Code[Int, nothing]
+
+:p
+`1 `+ `2 `* `3 `+ `4
+//│ |`|1| |`|+| |`|2| |`|*| |`|3| |`|+| |`|4|
+//│ AST: TypingUnit(Quoted(App(Var(+), Tup(_: Unquoted(Quoted(App(Var(+), Tup(_: Unquoted(Quoted(IntLit(1))), _: Unquoted(Quoted(App(Var(*), Tup(_: Unquoted(Quoted(IntLit(2))), _: Unquoted(Quoted(IntLit(3))))))))))), _: Unquoted(Quoted(IntLit(4)))))))
+//│ Parsed: code"+(${code"+(${code"1"}, ${code"*(${code"2"}, ${code"3"},)"},)"}, ${code"4"},)";
+//│ Code[Int, nothing]
+
+:e
+:p
+`1 `+ `2 + 3
+//│ |`|1| |`|+| |`|2| |+| |3|
+//│ AST: TypingUnit(App(Var(+), Tup(_: Quoted(App(Var(+), Tup(_: Unquoted(Quoted(IntLit(1))), _: Unquoted(Quoted(IntLit(2)))))), _: IntLit(3))))
+//│ Parsed: +(code"+(${code"1"}, ${code"2"},)", 3,);
+//│ ╔══[ERROR] Type mismatch in application:
+//│ ║  l.60: 	`1 `+ `2 + 3
+//│ ║        	^^^^^^^^^^^^
```

Comment:
The location for that type provenance is wrong. You probably used "term.toLoc" somewhere instead of the appropriate location.